src/Resolution.ts

```diff
@@ -559,8 +575,35 @@ export default class Resolution {
     return records;
   }
 
+  private static async getNetwork(
+    type: NamingServiceName.CNS | NamingServiceName.ENS,
+    config?: {url: string} | {provider: Provider}
+  ): Promise<{network: string, provider: Provider} | undefined> {
+    if (!config) {
+      return undefined;
+    }
+    const provider = hasProvider(config) ? config.provider : new FetchProvider(type, config.url);
+    try {
+      const network = await provider.request({method: "net_version"}) as number;
+      return {
+        network: EthereumNamingService.NetworkIdMap[network],
+        provider
+      };
+    } catch(error) {
+      // We want to communicate that config.url is not responding instead of NamingServiceDown from our FetchProvider.
+      if (error instanceof ResolutionError && error.code === ResolutionErrorCode.NamingServiceDown) {
+        throw new ResolutionError(
+          ResolutionErrorCode.NetworkNotFound,
```

Comment:
Suppose you have your DB down in your project. In this case, you don't want your DB query to return `UserNotFound` you'd better want to see that database is down. So I don't think that catching `NamingServiceDown` and rethrowing is good idea in this case.

Reply:
I agree. I've removed all catching like this in Resolution 2.0

Reply:
@JohnnyJumper I wouldn't catch error on that level except in case if network is not supported by Unstoppable